src/vs/base/browser/ui/table/tableWidget.ts

```diff
@@ -198,6 +199,7 @@ export class Table<TRow> implements ISpliceable<TRow>, IDisposable {
 		_options?: ITableOptions<TRow>
 	) {
 		this.domNode = append(container, $(`.monaco-table.${this.domId}`));
+		this.columns = columns;
```

Comment:
Could save this and the private line by adding `private columns` to the ctor, similar to `virtualDelegate`?